Update Readme file with more details for better on-boarding (#142)
gemwalletcom/core: README.md

[![Clippy check](https://github.com/gemwalletcom/core/actions/workflows/lint.yml/badge.svg)](https://github.com/gemwalletcom/core/actions/workflows/lint.yml)
[![Gemstone CI](https://github.com/gemwalletcom/core/actions/workflows/ci-gemstone.yml/badge.svg)](https://github.com/gemwalletcom/core/actions/workflows/ci-gemstone.yml)

# Gem Core Library

> The Gem Core Library is a Rust monorepo utilized within the Gem wallet. The Android and iOS versions of the wallet utilize this core as backend infrastructure, performing tasks such as push notifications, pricing, assets information, and more.

Reviewer: ```suggestion
The Gem Core library is a Rust monorepo that serves as the backend infrastructure and cross platform libraries for the Gem Wallet on Android and iOS. It handles various tasks, including:

- Push notifications
- Pricing information
- Asset updating
- Shared modes using TypeShare
- Crypto primitives
- More
```

## Gemstone 

The Gemstone is a cross-platform library for Swift and Kotlin.

## Cryptography

> The cryptography for the Gem wallet is implemented using [trust-wallet-core](https://github.com/trustwallet/wallet-core).

Reviewer: ```suggestion
The underlying cryptography / keystore for Gem Wallet is implemented using [Trust Wallet Core](https://github.com/trustwallet/wallet-core) for now.
```

```mermaid
graph LR

>     client[gem client like: swift or kotlin]

Reviewer: ```suggestion
    client[Gem Wallet (iOS/Android)]
```

>     core-lib[gem wallet core library monorepo]

Reviewer: ```suggestion
    core-lib[Gem Wallet Core]
```

>     trust-core[trust wallet core]

Reviewer: ```suggestion
    trust-core[Trust Wallet Core]
```
